packages/typespec-python/src/types.ts

```diff
@@ -173,21 +197,42 @@ function visibilityMapping(visibility?: Visibility[]): string[] | undefined {
     return result;
 }
 
+function createMultiPartFileType(type: SdkType): MultiPartFileType {
+    return { kind: "multipartfile", type };
+}
+
+function addDisableGenerationMap(type: SdkType): void {
+    if (disableGenerationMap.has(type)) return;
+
+    disableGenerationMap.add(type);
+    if (type.kind === "model" && type.baseModel) {
+        addDisableGenerationMap(type.baseModel);
+    } else if (type.kind === "array") {
+        addDisableGenerationMap(type.valueType);
+    }
+}
+
 function emitProperty<TServiceOperation extends SdkServiceOperation>(
     context: PythonSdkContext<TServiceOperation>,
     property: SdkBodyModelPropertyType,
 ): Record<string, any> {
+    const isMultipartFileInput = property.multipartOptions?.isFilePart;
+    const sourceType = isMultipartFileInput ? createMultiPartFileType(property.type) : property.type;
+    if (isMultipartFileInput) {
+        // Python convert all the type of file part to FileType so clear these models' usage so that they won't be generated
+        addDisableGenerationMap(property.type);
```

Comment:
no need to track, just skip the type generation.

Reply:
see https://github.com/Azure/autorest.python/pull/2740/files#r1721541783